cmd/tusd/cli/hooks.go

```diff
@@ -67,14 +70,67 @@ func invokeHookSync(typ HookType, info tusd.FileInfo, captureOutput bool) ([]byt
 		logEv("UploadTerminated", "id", info.ID)
 	}
 
-	if !Flags.HooksInstalled {
+	if !Flags.FileHooksInstalled && !Flags.HttpHooksInstalled {
 		return nil, nil
 	}
-
 	name := string(typ)
 	logEv("HookInvocationStart", "type", name, "id", info.ID)
 
-	cmd := exec.Command(Flags.HooksDir + "/" + name)
+	output := []byte{}
+	err := error(nil)
+
+	if Flags.FileHooksInstalled {
+		output, err = invokeFileHook(name, typ, info, captureOutput)
+	}
+
+	if Flags.HttpHooksInstalled {
+		output, err = invokeHttpHook(name, typ, info, captureOutput)
+	}
+
+	return output, err
+}
+
+func invokeHttpHook(name string, typ HookType, info tusd.FileInfo, captureOutput bool) ([]byte, error) {
+	url := Flags.HttpHooksEndpoint
+	jsonInfo, err := json.Marshal(info)
+
+	req, err := http.NewRequest("POST", url, bytes.NewReader(jsonInfo))
+	req.Header.Set("Upload-State", name)
```

Comment:
Hook-Name would be are more suited and understandable header name than Upload-State, IMO.